### PR TITLE
fix: local image bad quality [WPB-1839]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-1839" title="WPB-1839" target="_blank">BugWPB-1839</a>  [Import to Wire Dialog] Preview thumbnail on top is misaligned
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Image preview in ImportMediaScreen had bad quality

### Solutions

- removed HorizontalPage which was not good to use with images width smaller than screen
- extracted LocalImageAsset from painter causing bad image quality 
- added persistent list of assets for better performance
- used `AsyncImage` in `ImportedImageMessage` to simplify handling local assets

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [4252](https://10.10.124.134/job/AR-build-pipeline/4252/) **failed**.

**Comment by AndroidBob** [comment]
Build [4263](https://10.10.124.134/job/AR-build-pipeline/4263/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9437969)(9437969)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9437969)(9437969)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4271](https://10.10.124.134/job/AR-build-pipeline/4271/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9438712)(9438712)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9438712)(9438712)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/model/ImageAsset.kt [hunk 1]
```diff
@@ -30,35 +30,43 @@ import com.wire.kalium.logic.data.user.UserAssetId
...
     @Stable
-    data class LocalImageAsset(
-        private val imageLoader: WireSessionImageLoader,
+    data class Local(
         val dataPath: Path,
         val idKey: String
-    ) : ImageAsset(imageLoader) {
+    ) : ImageAsset()
 
+    sealed class Network(private val imageLoader: WireSessionImageLoader) : ImageAsset() {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/model/ImageAsset.kt]: nitpick: maybe `Remote` would be a better name than `Network`, fits better with `Local` 😄

**Review by saleniuk — APPROVED** [review]

**Review by yamilmedina — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4291](https://10.10.124.134/job/AR-build-pipeline/4291/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9446518)(9446518)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9446518)(9446518)-staging-compat.apk)

**Pull request merged**